tools/tpm2_stirrandom.c

```diff
@@ -0,0 +1,149 @@
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <ctype.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <tss2/tss2_esys.h>
+
+#include "log.h"
+#include "files.h"
+#include "tpm2_options.h"
+#include "tpm2_tool.h"
+#include "tpm2_util.h"
+
+/* Spec enforce input data to be not longer than 128 bytes */
+#define MAX_SIZE_TO_READ 128
+
+typedef struct tpm_stirrandom_ctx tpm_stirrandom_ctx;
+struct tpm_stirrandom_ctx {
+    TPM2B_SENSITIVE_DATA    in_data;
+    char*                   in_file;
+};
+
+static tpm_stirrandom_ctx ctx;
+
+static bool do_stirrandom(ESYS_CONTEXT *ectx) {
+    TSS2_RC rval = Esys_StirRandom(ectx, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE, &ctx.in_data);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_ERR("Error while injecting specified additionnal data into TPM random pool");
+        LOG_PERR(Esys_StirRandom, rval);
+        return false;
+    }
+
+    return true;
+}
+
+static bool on_args(int argc, char **argv) {
+
+    if (argc != 1) {
+        LOG_ERR("Only supports one FILE_INPUT file, got %d arguments", argc);
+        return false;
+    }
+
+    ctx.in_file = argv[0];
+
+    return true;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+
+    memset(&ctx, 0, sizeof(tpm_stirrandom_ctx));
+
+    *opts = tpm2_options_new(NULL, 0, NULL, NULL, on_args, 0);
+
+    return *opts != NULL;
+}
+
+static bool load_sensitive(void) {
+    FILE* file;
+
+    ctx.in_data.size = MAX_SIZE_TO_READ;
+
+    if(ctx.in_file){
+        unsigned long file_size;
+
+        LOG_INFO("Reading from file");
+        file = fopen(ctx.in_file, "rb");
+        if (!file) {
+            LOG_ERR("Could not open file \"%s\" error %s", ctx.in_file, strerror(errno));
+            return false;
+        }
+        if(!files_get_file_size(file, &file_size, ctx.in_file)){
+            goto err;
+        }
+
+        if(file_size > MAX_SIZE_TO_READ){
+            LOG_ERR("Input file is too big. Expected a maximum of 128 bytes, got : %ld", file_size);
+            goto err;
+        }
+    } else {
+        LOG_INFO("Reading from stdin");
+        file = stdin;
+    }
+
+    ctx.in_data.size = fread(&ctx.in_data.buffer, 1, ctx.in_data.size, file);
+    if (ferror(file)) {
+        LOG_ERR("Error while reading data");
+        goto err;
+    } 
+
+    if(ctx.in_data.size == 0){
+        LOG_ERR("Data size to send is zero");
+        goto err;
+    }
+
+    LOG_INFO("Submitting %d bytes to TPM", ctx.in_data.size);
+
+    return true;
+err:
+    if(ctx.in_file)
```

Comment:
brackets per coding standard

Reply:
Good catch !

Is it wanted to have CI perform a linting step ? I may open an issue on that and build it using a tool like astyle or clang-format.

Reply:
Yeah that would be great